src/pal/src/file/directory.cpp

```diff
@@ -255,9 +263,13 @@ RemoveDirectoryW(
         dwLastError = ERROR_PATH_NOT_FOUND;
         goto done;
     }
-    
-    mb_size = WideCharToMultiByte( CP_ACP, 0, lpPathName, -1, mb_dir, MAX_LONGPATH,
+
+    length = (PAL_wcslen(lpPathName)+1) * sizeof(WCHAR);
```

Comment:
No, we are getting the estimate from the number of WCHARs and for 4 byte encoding, the character is encoded as two WCHARS. So for 4 byte encoding, the number of encoded bytes the same as the sum of the input character sizes.
Example:
Unicode character with code 0x10000 is encoded as two WCHARS: 0xD800, 0xDC00
In UTF-8, it is encoded  as 0xF0,0x80,0x80,0x80